python/run.py

```diff
@@ -0,0 +1,168 @@
+HOME = os.getenv('HOME', '')    # HOME environment variable
+
+
+def print_if(cond, *args):
+    if (cond):
+        print(*args)
+
+
+def call_subprocess(process, keyword_args, trail_args):
+    """wrapper function to format kwargs into bash arguments, print and then make the call"""
+    single_char_options = [k for k, v in keyword_args.items() if v and len(k) == 1]
+    multiple_char_options = [k for k, v in keyword_args.items() if v and len(k) > 1]
+    opts = ["-{}{}".format(k.replace("_", "-"), keyword_args[k]) for k in single_char_options]
+    opts += ["--{}={}".format(k.replace("_", "-"), keyword_args[k]) for k in multiple_char_options]
+    print("calling subprocess: {}".format([process, ] + opts + list(trail_args)))
+    return subprocess.call([process, ] + opts + list(trail_args))
+
+
+@argh.arg("args", help="""list of files,packages or modules. if nothing is specified,
+default value is sparkdl""")
+def pylint(rcfile="./python/.pylint/accepted.rc", reports="y", confidence=None, disable=None,
+           enable=None, errors_only=None, evaluation=None, extension_pkg_whitelist=None,
+           full_documentation=None, generate_rcfile=None, help_msg=None, ignore=None,
+           ignore_patterns=None, init_hook=None, jobs=None, list_conf_levels=None,
+           list_msgs=None, load_plugins=None, long_help=None, msg_template=None,
+           output_format=None, persistent=None, py3k=None, score=None, suggestion_mode=None,
+           version=None, *args):
+    """
+    Wraps `pylint` and provides defaults. Trailing arguments are a list of files, packages or
+    modules. if nothing is specified, default value is ./python/sparkdl
+    """
+    if not args:
+        args = ("./python/sparkdl", )
+    kwargs = {k: v for k, v in locals().items() if k != "args" and v}
+    kwargs["m"] = "pylint"
+    return call_subprocess("python", keyword_args=kwargs, trail_args=args)
+
+
+def prospector(without_tool="pylint", *args):
+    """
+    Wraps `prospector` and provides defaults. Run `prospector --help` for more details. Trailing
+    arguments are a list of files, packages or modules. if nothing is specified,
+    default value is ./python/sparkdl
+    """
+    if not args:
+        args = ("./python/sparkdl", )
+    kwargs = {k: v for k, v in locals().items() if k != "args" and v}
+    kwargs["m"] = "prospector"
+    return call_subprocess("python", keyword_args=kwargs, trail_args=args)
+
+
+def yapf(style="{based_on_style=pep8, COLUMN_LIMIT=100}", in_place=False, recursive=False, *args):
+    """Wraps `yapf` and provides some defaults. Run `yapf --help` for more details."""
+    if in_place:
+        args = ("-i",) + args
+    if recursive:
+        args = ("-r",) + args
+    return call_subprocess("python", keyword_args={"m": "yapf", "style": style}, trail_args=args)
+
+
+def envsetup(default=False, interactive=False, missing_only=False, completion=False, verbose=False):
```

Comment:
How do you feel about having this function return a dictionary that we could pass to `subprocess.call`? We could also add a command option to print this ENV in a nice way for debugging.